engines/python/src/main/java/ai/djl/python/engine/Connection.java

```diff
@@ -126,12 +126,23 @@ static String[] getPythonStartCmd(
         int deviceId = device.getDeviceId();
         int clusterSize = PyEnv.getClusterSize();
         int tensorParallelDegree = pyEnv.getTensorParallelDegree();
+        int pipelineParallelDegree = pyEnv.getPipelineParallelDegree();
         String entryPoint = pyEnv.getEntryPoint();
         String recommendedEntryPoint = pyEnv.getRecommendedEntryPoint();
 
         if (PyEnv.isMultiNode()) {
-            String cudaDevices = getVisibleDevices(workerId, tensorParallelDegree / clusterSize);
+
+            int mpiSlots = (tensorParallelDegree*pipelineParallelDegree) / 2;
+            int mpiProcesses = tensorParallelDegree*pipelineParallelDegree;
+
+            // if (pipelineParallelDegree != clusterSize) {
```

Comment:
Remove commented out code.
We should check if total slots is divisible by clusterSize